wdl-analysis/src/types/v1.rs

```diff
@@ -1004,6 +1046,472 @@ where
+                        name.span(),
+                        expr_ty,
+                        expr.span(),
+                    ));
+                }
+
+                true
+            }
+            "fpga" if self.version >= SupportedVersion::V1(V1::Two) => {
+                if !expr_ty.is_coercible_to(self.types, &PrimitiveTypeKind::Boolean.into()) {
+                    self.diagnostics.push(type_mismatch(
+                        self.types,
+                        PrimitiveTypeKind::Boolean.into(),
+                        name.span(),
+                        expr_ty,
+                        expr.span(),
+                    ));
+                }
+
+                true
+            }
+            "disks" => {
+                if !expr_ty.is_coercible_to(self.types, &PrimitiveTypeKind::Integer.into())
+                    && !expr_ty.is_coercible_to(self.types, &PrimitiveTypeKind::String.into())
+                    && !expr_ty.is_coercible_to(self.types, &STDLIB.array_string)
+                {
+                    self.diagnostics.push(type_mismatch_custom(
+                        self.types,
+                        "type `Int`, type `String`, or type `Array[String]`",
+                        name.span(),
+                        expr_ty,
+                        expr.span(),
+                    ));
+                }
+
+                true
+            }
+            "max_retries" if self.version >= SupportedVersion::V1(V1::Two) => {
+                if !expr_ty.is_coercible_to(self.types, &PrimitiveTypeKind::Integer.into()) {
+                    self.diagnostics.push(type_mismatch(
+                        self.types,
+                        PrimitiveTypeKind::Integer.into(),
+                        name.span(),
+                        expr_ty,
+                        expr.span(),
+                    ));
+                }
+
+                true
+            }
+            "maxRetries" => {
+                if !expr_ty.is_coercible_to(self.types, &PrimitiveTypeKind::Integer.into()) {
+                    self.diagnostics.push(type_mismatch(
+                        self.types,
+                        PrimitiveTypeKind::Integer.into(),
+                        name.span(),
+                        expr_ty,
+                        expr.span(),
+                    ));
+                }
+
+                true
+            }
+            "return_codes" if self.version >= SupportedVersion::V1(V1::Two) => {
+                if !expr_ty.is_coercible_to(self.types, &PrimitiveTypeKind::Integer.into())
+                    && !expr_ty.is_coercible_to(self.types, &PrimitiveTypeKind::String.into())
+                    && !expr_ty.is_coercible_to(self.types, &STDLIB.array_int)
+                {
+                    self.diagnostics.push(type_mismatch_custom(
+                        self.types,
+                        "type `Int`, type `String`, or type `Array[Int]`",
```

Comment:
If this can be a `String` I would also expect `Array[String]` to be supported.

Reply:
The only supported string value is `*` to indicate any return code is treated as a success.

https://github.com/openwdl/wdl/blob/wdl-1.2/SPEC.md#return_codes

Reply:
I'll add a check for a value other than `*`.

Reply:
Actually, that's not something that can be statically validated as I forgot these are expressions and not strictly literals. Validation of the string will need to be done at evaluation time.

Reply:
Ok that makes sense 👍